### DISCUSSION TITLE
How to set priority for scenarios in the feature file

### DISCUSSION THREAD

**Discussion by pritesh848** [reqnroll/Reqnroll#289] [Contributor Q&A] [answered]
I have a feature with multiple scenarios and each scenario is linked to each others so I need to give priority to each scenario, any ways to set priority in reqnroll with xunit

@priority:1
Scenario: First scenario to run
...

@priority:2
Scenario: Second scenario to run
...

if I add this @priority:1 or @Priority1 nothing is happening it was taking scenario order of reverse alphanumeric of scenario first character

**Comment by clrudolphi** [comment]
The short answer is no.

Scenarios are (by default) executed sequentially in the order they are defined in the feature file. The only exception to that will come in the future as the Reqnroll team is working on support for Scenario level parallelization (in which case the scenarios will be executed simultaneously).

If you have Scenario 2 that shares some set-up that is completed in Scenario 1, have you considered using a [Background](https://cucumber.io/docs/gherkin/reference/#background)?

**Comment by gasparnagy** [comment]
In addition to what @clrudolphi mentioned, it is important to highlight that it is not Reqnroll who decides on the order how the tests are executed, but the test runner framework you use (e.g. MsTest), so if you really need to manipulate the execution order (which is not a good practice), you should check the options provided by your test runner tool, but there are generally no easy options for that.

**Reply by pritesh848** [reply]
Thanks for answering, @gasparnagy , @clrudolphi @q42jaap 
Yes I agree  Reqnroll not decide on the order in how the tests are executed, but the test runner framework you use (e.g. MsTest)
But whatever ordering is  supported by **xunit** is method based so if I use it in step-definition I need to use [Fact] so it was making it not in the BDD approach
and In mobile automation testing launching the app is too much time taking so adding app launch in the beforetestrun hooks and to execute all test cases needed to the ordering.
 (as other frameworks testng, junit, xunit, mstest atleast the configuration of ordering with reqnroll is required)

**Reply by ajeckmans** [reply, marked as answer]
If you start the app in beforetestrun and then on afterscenario make sure to reset its state and run everything sequentially (so not parallel) the test run order should not matter.

It is generally a bad practice to make scenarios depend on each other as they're then typically not really describing a scenario anymore.

If you really want to go down this path you've taken (and I'd seriously reconsider doing so) you can take a look at https://hamidmosalla.com/2018/08/16/xunit-control-the-test-execution-order/ there's a way to plug in this attribute through our plugin system.

Btw the steps should not have the Fact attribute as the steps merely describe steps in the final tests and are not tests by themselves. The Test methods are generated by ReqnRoll based on the scenarios and utilize/call the steps you've provided. So you need to manipulate the test generation itself through our plugin system.

Or a more simple approach would be to just start the scenario name with a number and control the ordering that way.

**Reply by pritesh848** [reply]
@ajeckmans  thanks for your recommendations 
Yes I reconsidered it and removed adding the sequence for scenario execution.
In my old testNG framework I used it to as one scenario making some test data and another scenario performing action on it, so now we resolved that dependency by adding dee-links for mobile automation screen so navigation time is saved and dependency was removed 
Thanks for your continuous quick response

**Comment by pritesh848** [comment]
I am trying to run feature files with tagged as @RO using cmd command
 dotnet test MobileAutomation.sln --filter "Traits=RO"

is it possible to run like this in  reqnroll ?

getting below error
Starting test execution, please wait...
Logging Vstest Diagnostics in file: C:\MobileAutomationBDD\Mobile-Automation
A total of 1 test files matched the specified pattern.
No test matches the given testcase filter `Traits=RO` in C:\MobileAutomationBDD\Mobile-Automation\bin\Debug\net8.0\Mobile.dll

**Reply by gasparnagy** [reply]
Please refer to https://learn.microsoft.com/en-us/dotnet/core/testing/selective-unit-tests?pivots=mstest

The filter syntax depends on your test execution framework!
